test/s3select_test.cpp

```diff
@@ -2603,3 +2655,47 @@ TEST(TestS3selectFunctions, nested_query_multirow_result)
   s3select_res = run_s3select(input_query, input_csv);
   EXPECT_EQ(s3select_res, expected_res);
 }
+
+TEST(TestS3selectFunctions, opserialization_expressions)
+{
+  std::string input;
+  size_t size = 10;
+  generate_rand_columns_csv(input, size);
+
+  char a[5] = {'@', '#', '$', '%'};
+  char b[4] = {'!', '^', '&', '*'};
+  char x = a[rand() % 4];
+  char y = b[rand() % 4];
+
+  const std::string input_query = "select * from s3object ;";
+
+  run_s3select_test_opserialization(input_query, input, &x, &y);
+
+  const std::string input_query_1 = "select 1 + 2 from s3object;";
+
+  std::string s3select_result_1 = run_s3select_opserialization_quot(input_query_1,input, true);
+
+  const std::string input_query_2 = "select 7 - 4 from s3object;";
+
+  std::string s3select_result_2 = run_s3select_opserialization_quot(input_query_2,input, true);
+
+  ASSERT_EQ(s3select_result_1, s3select_result_2);
```

Comment:
the test comparing 2 constant expressions ( 1+2 against 7-4), is not a useful test.
it's better to use dynamic ( a - b = b - a ... etc )

another thing.
this feature is about converting the result into different CSV settings;
testing single row/single column is not covering much.

as in the previous test, it should work on multiple columns and multiple rows.